custom_components/tuya_local/devices/yym_805SW_aroma_nightlight.yaml

```diff
@@ -7,18 +8,114 @@ primary_entity:
     - id: 1
       name: switch
       type: boolean
+      mapping:
+        - dps_val: true
+          icon: mdi:scent
+          value: ON
+        - dps_val: false
+          icon: mdi:scent-off
+          value: OFF      
     - id: 2
       name: speed
       type: string
+      hidden: true
       mapping:
         - dps_val: large
           value: 100
+          icon: "mdi:sprinkler-variant"
         - dps_val: small
           value: 50
+          icon: "mdi:sprinkler"
+    - id: 4          
+      name: "Left time"
+      readolny: true
+      optional: true
+      type: integer
+      range:
+        min: 0
+        max: 360
 secondary_entities:
   - entity: light
     name: Nightlight
+    icon: "mdi:light-recessed"
     dps:
       - id: 5
         name: switch
         type: boolean
+      - id: 6 
```

Comment:
Not really needed if a separate select is used.  Alternatively you could move the "colourful1" here, then select it as an effect on the light instead of a separate select..

Reply:
When it is in effect under light it doesn't change. Having separated select works fine.
I don't know if it is related to supported mode of light because hv is not supported - on Tuya integration mode is hv: supported_color_modes: hs.